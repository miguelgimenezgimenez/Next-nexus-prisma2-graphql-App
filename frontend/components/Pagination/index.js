import React from 'react';
import Head from 'next/head';
import Link from 'next/link';
import PaginationStyles from './styles';
import { PER_PAGE } from '../../constants';


function Pagination({ page, pathname, totalCount, brand_id }) {

  const pages = Math.ceil(totalCount / PER_PAGE);
  return (
    <PaginationStyles data-testid="pagination">
      <Head>
        <title>
          Phones  Page {page}
        </title>
      </Head>
      <Link

        href={{
          pathname: pathname,
          query: { page: page - 1, brand_id },
        }}
      >
        <a className="prev" aria-disabled={page <= 1}>
          ← Prev
        </a>
      </Link>
      <p>
        Page {page} of{' '}
        <span className="totalPages" data-testid="totalPages">
          {pages}
        </span>
      </p>
      <p>{totalCount} Items Total</p>
      <Link

        href={{
          pathname: pathname,
          query: { page: page + 1, brand_id },
        }}
      >
        <a className="next" aria-disabled={page >= pages}>
          Next →
        </a>
      </Link>
    </PaginationStyles>
  );
}

export default Pagination;

