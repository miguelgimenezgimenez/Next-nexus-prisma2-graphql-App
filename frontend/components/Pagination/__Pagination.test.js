import { render } from '@testing-library/react';
const { perPage } = require('../../config')
import Pagination, { PAGINATION_QUERY } from './index';

describe('<Pagination/>', () => {

  it('renders pagination for 18 items', () => {
    const { container, findByTestId, getByTestId, debug, query } = render(
      <Pagination page={1} totalCount={18} />
    );

    expect(getByTestId('totalPages')).toHaveTextContent('5');
    expect(container).toMatchSnapshot();
  });

  it('disables prev button on first page', () => {
    const { getByText, findByTestId } = render(
      <Pagination page={1} totalCount={20} />
    );
    expect(getByText(/Prev/)).toHaveAttribute('aria-disabled', 'true');
    expect(getByText(/Next/)).toHaveAttribute('aria-disabled', 'false');
  });

  it('disables next button on last page', () => {
    const { getByText, findByTestId } = render(
      <Pagination page={5} totalCount={5 * perPage} />
    );
    expect(getByText(/Prev/)).toHaveAttribute('aria-disabled', 'false');
    expect(getByText(/Next/)).toHaveAttribute('aria-disabled', 'true');
  });

  it('enables all buttons on a middle page', () => {
    const { getByText, findByTestId } = render(
      <Pagination page={3} totalCount={20} />
    );
    expect(getByText(/Prev/)).toHaveAttribute('aria-disabled', 'false');
    expect(getByText(/Next/)).toHaveAttribute('aria-disabled', 'false');
  });
});
