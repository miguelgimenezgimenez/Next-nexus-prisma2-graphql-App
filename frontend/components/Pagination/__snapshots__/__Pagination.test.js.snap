// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> renders pagination for 18 items 1`] = `
<div>
  <div
    class="styles__PaginationStyles-sc-1b98u0p-0 kIHkST"
    data-testid="pagination"
  >
    <a
      aria-disabled="true"
      class="prev"
      href="/?page=0&brand_id="
    >
      ← Prev
    </a>
    <p>
      Page 
      1
       of
       
      <span
        class="totalPages"
        data-testid="totalPages"
      >
        5
      </span>
    </p>
    <p>
      18
       Items Total
    </p>
    <a
      aria-disabled="false"
      class="next"
      href="/?page=2&brand_id="
    >
      Next →
    </a>
  </div>
</div>
`;
